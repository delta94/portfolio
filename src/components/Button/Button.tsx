import React from 'react';
import { createUseStyles } from 'react-jss';
import { CustomTheme } from '../../theme';

const useStyles = createUseStyles((theme: CustomTheme) => ({
  mainContainer: {
    marginTop: theme.spacing(4),
    backgroundColor: theme.colors.thirdColor,
    color: theme.colors.white,
    borderWidth: 0,
    borderRadius: 25,
    maxWidth: 172,
    width: '100%',
    maxHeight: 40,
    height: '100%',
    outline: 'none',
    cursor: 'pointer',
    '&:hover': {
      backgroundColor: theme.colors.darkerThirdColor,
    },
  },
}));

interface ButtonProps {}

const Button: React.FC<ButtonProps> = ({ children }) => {
  const classes: Record<'mainContainer', string> = useStyles();

  return (
    <button onClick={() => window.open('mailto:sk.zeeshan1992@gmail.com', '_blank')} className={classes.mainContainer}>
      {children}
    </button>
  );
};

export { Button };
