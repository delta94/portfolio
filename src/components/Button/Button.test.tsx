import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { Button } from './Button';

test('Button renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <Button>Mock Button</Button>
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
