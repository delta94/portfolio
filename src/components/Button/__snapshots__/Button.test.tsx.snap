// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<DocumentFragment>
  <button
    class="mainContainer-0-2-1"
    style="transform: perspective(600px) rotateX(0deg) rotateY(0deg) scale(1);"
  >
    Mock Button
  </button>
</DocumentFragment>
`;
